packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -374,16 +375,30 @@ export class GutenbergEditorPage {
 
 		await frame.click( selectors.publishButton( selectors.postToolbar ) );
 		await frame.click( selectors.publishButton( selectors.publishPanel ) );
-
-		const viewPublishedArticleButton = await frame.waitForSelector( selectors.viewButton );
-		const publishedURL = ( await viewPublishedArticleButton.getAttribute( 'href' ) ) as string;
+		const publishedURL = await this.getPublishedURL();
 
 		if ( visit ) {
 			await this.visitPublishedPost( publishedURL );
 		}
 		return publishedURL;
 	}
 
+	/**
+	 * Obtains the published article's URL from post-publish panels.
+	 *
+	 * This method is only able to obtain the published article's URL if immediately
+	 * preceded by the action of publishing the article *and* the post-publish panel
+	 * being visible.
+	 *
+	 * @returns {Promise<string>} Published article's URL.
+	 */
+	async getPublishedURL(): Promise< string > {
+		const frame = await this.getEditorFrame();
+
+		const viewPublishedArticleButton = await frame.waitForSelector( selectors.viewButton );
+		return ( await viewPublishedArticleButton.getAttribute( 'href' ) ) as string;
+	}
```

Comment:
By moving functionality out to a stand-alone method, it decouples the action of retrieving the published article's URL from the act of publishing an article.

This is relevant here as when the `Private` visibility is chosen, the post is published immediately ie. without clicking on the `Publish` button.